src/utils/lock_wrapper.h

```diff
@@ -214,27 +212,23 @@ class lock_spin_recursive : public lock_spin
 	~lock_spin_recursive() {};
 
 	inline int lock() {
-#ifdef DEFINED_NO_THREAD_LOCK
-		return 0;
-#endif // DEFINED_NO_THREAD_LOCK
+		DEFINED_NO_THREAD_LOCK_RETURN
```

Comment:
@liranoz12 @OphirMunk is there probability that enabling DEFINED_NO_THREAD_LOCK forces compilation error for example 'Unreachable code'?